### ISSUE TITLE
Downloading thread support like 1dm 

### ISSUE THREAD

**Issue by Krishna-art-dev** [recloudstream/cloudstream#964]
### Describe your suggested feature

Make a setting which lets us choice thread to download movie anime more fastly

Because some plugins doesn't let us download movie fast like movierulzhd from hexa but threads can able to lets us download it more faster 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by fire-light42** [comment]
The downloader is already multithreaded?

**Comment by Krishna-art-dev** [comment]
> The downloader is already multithreaded?

Idk there no option or menu to see downloader is multi threaded or not 
Some extension gives a lot of speed 20-30mb on anichi 

While some extension just doesn't, like movierulzhd ( I think it's because it is a streaming site ) in streaming we need smaller packets of info / video so I think they restrict on above that and multi threaded download should help there 
Movierulzhd gives me around 2-3mbps download speed

And using 1dm with threads( packets download using 16 packet on 1dm ) around gives around 9-11mbps

**Comment by Krishna-art-dev** [comment]
@fire-light42

**Comment by fire-light42** [comment]
1. Dont ping
2. The download speed is dependent on the hosting site, and we already always use multi threading but limit it to 3 to not lag the phone or get blocked by the hosting site. The option to change this has not been implemented yet.

**Comment by Krishna-art-dev** [comment]
Oh ok thanks for the information
